lib/elixir/src/elixir_expand.erl

```diff
@@ -921,6 +922,28 @@ assert_generator_start(Meta, _, E) ->
 
 %% Assertions
 
+refute_parallel_bitstring_match({'<<>>', _, _}, {'<<>>', Meta, _} = Arg, E, true) ->
+  form_error(Meta, ?key(E, file), ?MODULE, {parallel_bitstring_match, Arg});
+refute_parallel_bitstring_match(Expr, {'=', _Meta, [Left, Right]}, E, Parallel) ->
+  refute_parallel_bitstring_match(Expr, Left, E, true),
+  refute_parallel_bitstring_match(Expr, Right, E, Parallel);
+refute_parallel_bitstring_match([_ | _] = Expr, [_ | _] = Arg, E, Parallel) ->
+  refute_parallel_bitstring_match_each(Expr, Arg, E, Parallel);
+refute_parallel_bitstring_match({Left1, Left2}, {Right1, Right2}, E, Parallel) ->
+  refute_parallel_bitstring_match_each([Left1, Left2], [Right1, Right2], E, Parallel);
+refute_parallel_bitstring_match({'{}', _, Args1}, {'{}', _, Args2}, E, Parallel) ->
+  refute_parallel_bitstring_match_each(Args1, Args2, E, Parallel);
+refute_parallel_bitstring_match({'%{}', _, Args1}, {'%{}', _, Args2}, E, Parallel) ->
+  refute_parallel_bitstring_match(Args1, Args2, E, Parallel);
```

Comment:
We should add a test for structs and also make sure they are checked. At this point they have a special notation so we need to check for `{%, _, [Name, Map]}`.